resources.management_groups.tf

```
@@ -76,6 +76,40 @@ resource "azurerm_management_group" "level_6" {
 
 }
 
+# This will deploy Diagnostic Settings for the Management Groups
+# when the input variable deploy_diagnostics_for_mg is true 
+resource "azapi_resource" "diag_settings" {
+  for_each  = toset(local.azapi_diagnostics)
```

Please can you change this to use the map instead?

```suggestion
  for_each  = local.azapi_mg_diagnostics
```